drivers/sx126x/sx126x.c

```diff
@@ -148,6 +166,7 @@ int sx126x_init(sx126x_t *dev)
 
     DEBUG("[sx126x] init: SPI_%i initialized with success\n", dev->params->spi);
 
+#if !defined(CPU_FAM_STM32WL)
```

Comment:
@leandrolanzieri Can you suggest the best method to do this ?

There are SOC specific code based on  `CPU_FAM_STM32WL` and Board specific ones based on `BOARD_NUCLEO_WL55JC`. Should this be a feature?

Reply:
From what I understand, the code is very CPU/SOC specific so hardcoding the CPU family I guess is okay. In the case of boards yes, the list could grow hence we need to find a better way.

Reply:
On an offline discussion with @jia200x we concluded that in order to make this more generic, a sort of splitting of the driver would be the best. Having an SPI-specific part and another that applies to SOCs that integrate the radio. When specifying the ISR vector clearly `ifdef`ing the family is fine, but it is oddly specific to use it to avoid the compilation of the GPIO IRQ functionalities. Better go generic here.

Reply:
the code here is not CPU specific, but it shouldn't be loaded in the STM32WL case. If there's another SOC that contains similar features, we will need to keep changing these files.

Even more, having these `#ifdef...#endif` everywhere makes it really hard to maintain. For this pattern, I would suggest to use common functions and implement them accordingly. E.g you could have a `sx126x_hw_init` that's implemented for the "SPI" version and the "Periph" version.